internal/controller/kyma/controller.go

```diff
@@ -584,3 +631,30 @@ func (r *Reconciler) SyncKymaEnabled(kyma *v1beta2.Kyma) bool {
 func (r *Reconciler) IsKymaManaged() bool {
 	return r.IsManagedKyma
 }
+
+// ReconciliationLoopResult is a struct that allows helper functions used in reconciliation process to indicate that the main reconciliation loop should be finished now using certain values.
+type ReconciliationLoopResult struct {
```

Comment:
As I explained, this data transfer struct is not useful, but makes the return logic hard to understand imo.

Reply:
update: I don't think you need to change this existing logic https://github.com/kyma-project/lifecycle-manager/pull/1748#discussion_r1711371963

Reply:
removed, reconcile logic reverted.